### PR TITLE
Input checks

### LINKED ISSUE #14: Issue when trying to connect to postgres cluster
When running the version nightly-20171010 and trying to connect to a postgres cluster, the following error is presented:

```
Waiting for postgres...
Postgres is ready
[2017-11-09 17:57:14 +0000] [1] [INFO] Starting gunicorn 19.6.0
[2017-11-09 17:57:14 +0000] [1] [INFO] Listening at: http://0.0.0.0:5000 (1)
[2017-11-09 17:57:14 +0000] [1] [INFO] Using worker: gevent
[2017-11-09 17:57:14 +0000] [16] [INFO] Booting worker with pid: 16
[2017-11-09 17:57:15 +0000] [16] [ERROR] Exception in worker process
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/SQLAlchemy-1.1.14-py3.6-linux-x86_64.egg/sqlalchemy/pool.py", line 1122, in _do_get
    return self._pool.get(wait, self._timeout)
  File "/usr/local/lib/python3.6/site-packages/SQLAlchemy-1.1.14-py3.6-linux-x86_64.egg/sqlalchemy/util/queue.py", line 145, in get
    raise Empty
sqlalchemy.util.queue.Empty

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/py_postgresql-1.2.1-py3.6-linux-x86_64.egg/postgresql/driver/pq3.py", line 2427, in connect
    self._establish()
  File "/usr/local/lib/python3.6/site-packages/py_postgresql-1.2.1-py3.6-linux-x86_64.egg/postgresql/driver/pq3.py", line 2553, in _establish
    self.typio.raise_client_error(could_not_connect, creator = self, cause = exc)
  File "/usr/local/lib/python3.6/site-packages/py_postgresql-1.2.1-py3.6-linux-x86_64.egg/postgresql/driver/pq3.py", line 514, in raise_client_error
    raise client_error
postgresql.exceptions.ClientCannotConnectError: could not establish connection to server
  CODE: 08001
  LOCATION: CLIENT
CONNECTION: [failed]
  failures[0]:
    SSL socket('10.107.62.14', 5432)
    Traceback (most recent call last):
      File "/usr/local/lib/python3.6/site-packages/SQLAlchemy-1.1.14-py3.6-linux-x86_64.egg/sqlalchemy/pool.py", line 1122, in _do_get
        return self._pool.get(wait, self._timeout)
      File "/usr/local/lib/python3.6/site-packages/SQLAlchemy-1.1.14-py3.6-linux-x86_64.egg/sqlalchemy/util/queue.py", line 145, in get
        raise Empty
    sqlalchemy.util.queue.Empty

    During handling of the above exception, another exception occurred:

    Traceback (most recent call last):
      File "/usr/local/lib/python3.6/site-packages/py_postgresql-1.2.1-py3.6-linux-x86_64.egg/postgresql/protocol/client3.py", line 184, in connect
        self.socket = self.socket_factory.secure(self.socket)
      File "/usr/local/lib/python3.6/site-packages/py_postgresql-1.2.1-py3.6-linux-x86_64.egg/postgresql/python/socket.py", line 56, in secure
        return ssl.wrap_socket(socket, **self.socket_secure)
      File "/usr/local/lib/python3.6/site-packages/gevent-1.1.1-py3.6-linux-x86_64.egg/gevent/_ssl3.py", line 576, in wrap_socket
        ciphers=ciphers)
      File "/usr/local/lib/python3.6/site-packages/gevent-1.1.1-py3.6-linux-x86_64.egg/gevent/_ssl3.py", line 126, in __init__
        self._context.verify_mode = cert_reqs
      File "/usr/local/lib/python3.6/ssl.py", line 485, in verify_mode
        super(SSLContext, SSLContext).verify_mode.__set__(self, value)
      File "/usr/local/lib/python3.6/ssl.py", line 485, in verify_mode
        super(SSLContext, SSLContext).verify_mode.__set__(self, value)
      File "/usr/local/lib/python3.6/ssl.py", line 485, in verify_mode
        super(SSLContext, SSLContext).verify_mode.__set__(self, value)
      [Previous line repeated 300 more times]
    RecursionError: maximum recursion depth exceeded

    The above exception was the direct cause of the following exception:

    postgresql.exceptions.InsecurityError: SSL negotiation caused exception
      CODE: --SEC
      LOCATION: CLIENT
CONNECTOR: [Host] pq://postgres:***@postgres:5432
  category: None
DRIVER: postgresql.driver.pq3.Driver

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/SQLAlchemy-1.1.14-py3.6-linux-x86_64.egg/sqlalchemy/engine/base.py", line 2147, in _wrap_pool_connect
    return fn()
  File "/usr/local/lib/python3.6/site-packages/SQLAlchemy-1.1.14-py3.6-linux-x86_64.egg/sqlalchemy/pool.py", line 387, in connect
    return _ConnectionFairy._checkout(self)
  File "/usr/local/lib/python3.6/site-packages/SQLAlchemy-1.1.14-py3.6-linux-x86_64.egg/sqlalchemy/pool.py", line 766, in _checkout
    fairy = _ConnectionRecord.checkout(pool)
  File "/usr/local/lib/python3.6/site-packages/SQLAlchemy-1.1.14-py3.6-linux-x86_64.egg/sqlalchemy/pool.py", line 516, in checkout
    rec = pool._do_get()
  File "/usr/local/lib/python3.6/site-packages/SQLAlchemy-1.1.14-py3.6-linux-x86_64.egg/sqlalchemy/pool.py", line 1138, in _do_get
    self._dec_overflow()
  File "/usr/local/lib/python3.6/site-packages/SQLAlchemy-1.1.14-py3.6-linux-x86_64.egg/sqlalchemy/util/langhelpers.py", line 66, in __exit__
    compat.reraise(exc_type, exc_value, exc_tb)
  File "/usr/local/lib/python3.6/site-packages/SQLAlchemy-1.1.14-py3.6-linux-x86_64.egg/sqlalchemy/util/compat.py", line 187, in reraise
    raise value
  File "/usr/local/lib/python3.6/site-packages/SQLAlchemy-1.1.14-py3.6-linux-x86_64.egg/sqlalchemy/pool.py", line 1135, in _do_get
    return self._create_connection()
  File "/usr/local/lib/python3.6/site-packages/SQLAlchemy-1.1.14-py3.6-linux-x86_64.egg/sqlalchemy/pool.py", line 333, in _create_connection
    return _ConnectionRecord(self)
  File "/usr/local/lib/python3.6/site-packages/SQLAlchemy-1.1.14-py3.6-linux-x86_64.egg/sqlalchemy/pool.py", line 461, in __init__
    self.__connect(first_connect_check=True)
  File "/usr/local/lib/python3.6/site-packages/SQLAlchemy-1.1.14-py3.6-linux-x86_64.egg/sqlalchemy/pool.py", line 651, in __connect
    connection = pool._invoke_creator(self)
  File "/usr/local/lib/python3.6/site-packages/SQLAlchemy-1.1.14-py3.6-linux-x86_64.egg/sqlalchemy/engine/strategies.py", line 105, in connect
    return dialect.connect(*cargs, **cparams)
  File "/usr/local/lib/python3.6/site-packages/SQLAlchemy-1.1.14-py3.6-linux-x86_64.egg/sqlalchemy/engine/default.py", line 393, in connect
    return self.dbapi.connect(*cargs, **cparams)
  File "/usr/local/lib/python3.6/site-packages/py_postgresql-1.2.1-py3.6-linux-x86_64.egg/postgresql/driver/dbapi20.py", line 409, in connect
    return driver.connect(**params)
  File "/usr/local/lib/python3.6/site-packages/py_postgresql-1.2.1-py3.6-linux-x86_64.egg/postgresql/driver/pq3.py", line 3037, in connect
    c.connect()
  File "/usr/local/lib/python3.6/site-packages/py_postgresql-1.2.1-py3.6-linux-x86_64.egg/postgresql/driver/dbapi20.py", line 354, in connect
    super().connect(*args, **kw)
  File "/usr/local/lib/python3.6/site-packages/py_postgresql-1.2.1-py3.6-linux-x86_64.egg/postgresql/driver/pq3.py", line 2430, in connect
    self.close()
  File "/usr/local/lib/python3.6/site-packages/py_postgresql-1.2.1-py3.6-linux-x86_64.egg/postgresql/driver/dbapi20.py", line 363, in close
    creator = self
postgresql.exceptions.Error: connection already closed
  LOCATION: CLIENT
CONNECTION: [failed]
  failures[0]:
    SSL socket('10.107.62.14', 5432)
    Traceback (most recent call last):
      File "/usr/local/lib/python3.6/site-packages/SQLAlchemy-1.1.14-py3.6-linux-x86_64.egg/sqlalchemy/pool.py", line 1122, in _do_get
        return self._pool.get(wait, self._timeout)
      File "/usr/local/lib/python3.6/site-packages/SQLAlchemy-1.1.14-py3.6-linux-x86_64.egg/sqlalchemy/util/queue.py", line 145, in get
        raise Empty
    sqlalchemy.util.queue.Empty

    During handling of the above exception, another exception occurred:

    Traceback (most recent call last):
      File "/usr/local/lib/python3.6/site-packages/py_postgresql-1.2.1-py3.6-linux-x86_64.egg/postgresql/protocol/client3.py", line 184, in connect
        self.socket = self.socket_factory.secure(self.socket)
      File "/usr/local/lib/python3.6/site-packages/py_postgresql-1.2.1-py3.6-linux-x86_64.egg/postgresql/python/socket.py", line 56, in secure
        return ssl.wrap_socket(socket, **self.socket_secure)
      File "/usr/local/lib/python3.6/site-packages/gevent-1.1.1-py3.6-linux-x86_64.egg/gevent/_ssl3.py", line 576, in wrap_socket
        ciphers=ciphers)
      File "/usr/local/lib/python3.6/site-packages/gevent-1.1.1-py3.6-linux-x86_64.egg/gevent/_ssl3.py", line 126, in __init__
        self._context.verify_mode = cert_reqs
      File "/usr/local/lib/python3.6/ssl.py", line 485, in verify_mode
        super(SSLContext, SSLContext).verify_mode.__set__(self, value)
      File "/usr/local/lib/python3.6/ssl.py", line 485, in verify_mode
        super(SSLContext, SSLContext).verify_mode.__set__(self, value)
      File "/usr/local/lib/python3.6/ssl.py", line 485, in verify_mode
        super(SSLContext, SSLContext).verify_mode.__set__(self, value)
      [Previous line repeated 300 more times]
    RecursionError: maximum recursion depth exceeded

    The above exception was the direct cause of the following exception:

    postgresql.exceptions.InsecurityError: SSL negotiation caused exception
      CODE: --SEC
      LOCATION: CLIENT
CONNECTOR: [Host] pq://postgres:***@postgres:5432
  category: None
DRIVER: postgresql.driver.pq3.Driver

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/gunicorn-19.6.0-py3.6.egg/gunicorn/arbiter.py", line 557, in spawn_worker
    worker.init_process()
  File "/usr/local/lib/python3.6/site-packages/gunicorn-19.6.0-py3.6.egg/gunicorn/workers/ggevent.py", line 190, in init_process
    super(GeventWorker, self).init_process()
  File "/usr/local/lib/python3.6/site-packages/gunicorn-19.6.0-py3.6.egg/gunicorn/workers/base.py", line 126, in init_process
    self.load_wsgi()
  File "/usr/local/lib/python3.6/site-packages/gunicorn-19.6.0-py3.6.egg/gunicorn/workers/base.py", line 136, in load_wsgi
    self.wsgi = self.app.wsgi()
  File "/usr/local/lib/python3.6/site-packages/gunicorn-19.6.0-py3.6.egg/gunicorn/app/base.py", line 67, in wsgi
    self.callable = self.load()
  File "/usr/local/lib/python3.6/site-packages/gunicorn-19.6.0-py3.6.egg/gunicorn/app/wsgiapp.py", line 65, in load
    return self.load_wsgiapp()
  File "/usr/local/lib/python3.6/site-packages/gunicorn-19.6.0-py3.6.egg/gunicorn/app/wsgiapp.py", line 52, in load_wsgiapp
    return util.import_app(self.app_uri)
  File "/usr/local/lib/python3.6/site-packages/gunicorn-19.6.0-py3.6.egg/gunicorn/util.py", line 357, in import_app
    __import__(module)
  File "/usr/src/app/auth/webRoutes.py", line 15, in <module>
    import database.CRUDController as crud
  File "/usr/src/app/auth/database/CRUDController.py", line 9, in <module>
    from database.Models import Permission, User, Group, PermissionEnum
  File "/usr/src/app/auth/database/Models.py", line 10, in <module>
    from flaskAlchemyInit import app, db
  File "/usr/src/app/auth/flaskAlchemyInit.py", line 32, in <module>
    db.create_all()
  File "/usr/local/lib/python3.6/site-packages/Flask_SQLAlchemy-2.3.0-py3.6.egg/flask_sqlalchemy/__init__.py", line 946, in create_all
    self._execute_for_all_tables(app, bind, 'create_all')
  File "/usr/local/lib/python3.6/site-packages/Flask_SQLAlchemy-2.3.0-py3.6.egg/flask_sqlalchemy/__init__.py", line 938, in _execute_for_all_tables
    op(bind=self.get_engine(app, bind), **extra)
  File "/usr/local/lib/python3.6/site-packages/SQLAlchemy-1.1.14-py3.6-linux-x86_64.egg/sqlalchemy/sql/schema.py", line 3949, in create_all
    tables=tables)
  File "/usr/local/lib/python3.6/site-packages/SQLAlchemy-1.1.14-py3.6-linux-x86_64.egg/sqlalchemy/engine/base.py", line 1928, in _run_visitor
    with self._optional_conn_ctx_manager(connection) as conn:
  File "/usr/local/lib/python3.6/contextlib.py", line 81, in __enter__
    return next(self.gen)
  File "/usr/local/lib/python3.6/site-packages/SQLAlchemy-1.1.14-py3.6-linux-x86_64.egg/sqlalchemy/engine/base.py", line 1921, in _optional_conn_ctx_manager
    with self.contextual_connect() as conn:
  File "/usr/local/lib/python3.6/site-packages/SQLAlchemy-1.1.14-py3.6-linux-x86_64.egg/sqlalchemy/engine/base.py", line 2112, in contextual_connect
    self._wrap_pool_connect(self.pool.connect, None),
  File "/usr/local/lib/python3.6/site-packages/SQLAlchemy-1.1.14-py3.6-linux-x86_64.egg/sqlalchemy/engine/base.py", line 2151, in _wrap_pool_connect
    e, dialect, self)
  File "/usr/local/lib/python3.6/site-packages/SQLAlchemy-1.1.14-py3.6-linux-x86_64.egg/sqlalchemy/engine/base.py", line 1465, in _handle_dbapi_exception_noconnection
    exc_info
  File "/usr/local/lib/python3.6/site-packages/SQLAlchemy-1.1.14-py3.6-linux-x86_64.egg/sqlalchemy/util/compat.py", line 203, in raise_from_cause
    reraise(type(exception), exception, tb=exc_tb, cause=cause)
  File "/usr/local/lib/python3.6/site-packages/SQLAlchemy-1.1.14-py3.6-linux-x86_64.egg/sqlalchemy/util/compat.py", line 186, in reraise
    raise value.with_traceback(tb)
  File "/usr/local/lib/python3.6/site-packages/SQLAlchemy-1.1.14-py3.6-linux-x86_64.egg/sqlalchemy/engine/base.py", line 2147, in _wrap_pool_connect
    return fn()
  File "/usr/local/lib/python3.6/site-packages/SQLAlchemy-1.1.14-py3.6-linux-x86_64.egg/sqlalchemy/pool.py", line 387, in connect
    return _ConnectionFairy._checkout(self)
  File "/usr/local/lib/python3.6/site-packages/SQLAlchemy-1.1.14-py3.6-linux-x86_64.egg/sqlalchemy/pool.py", line 766, in _checkout
    fairy = _ConnectionRecord.checkout(pool)
  File "/usr/local/lib/python3.6/site-packages/SQLAlchemy-1.1.14-py3.6-linux-x86_64.egg/sqlalchemy/pool.py", line 516, in checkout
    rec = pool._do_get()
  File "/usr/local/lib/python3.6/site-packages/SQLAlchemy-1.1.14-py3.6-linux-x86_64.egg/sqlalchemy/pool.py", line 1138, in _do_get
    self._dec_overflow()
  File "/usr/local/lib/python3.6/site-packages/SQLAlchemy-1.1.14-py3.6-linux-x86_64.egg/sqlalchemy/util/langhelpers.py", line 66, in __exit__
    compat.reraise(exc_type, exc_value, exc_tb)
  File "/usr/local/lib/python3.6/site-packages/SQLAlchemy-1.1.14-py3.6-linux-x86_64.egg/sqlalchemy/util/compat.py", line 187, in reraise
    raise value
  File "/usr/local/lib/python3.6/site-packages/SQLAlchemy-1.1.14-py3.6-linux-x86_64.egg/sqlalchemy/pool.py", line 1135, in _do_get
    return self._create_connection()
  File "/usr/local/lib/python3.6/site-packages/SQLAlchemy-1.1.14-py3.6-linux-x86_64.egg/sqlalchemy/pool.py", line 333, in _create_connection
    return _ConnectionRecord(self)
  File "/usr/local/lib/python3.6/site-packages/SQLAlchemy-1.1.14-py3.6-linux-x86_64.egg/sqlalchemy/pool.py", line 461, in __init__
    self.__connect(first_connect_check=True)
  File "/usr/local/lib/python3.6/site-packages/SQLAlchemy-1.1.14-py3.6-linux-x86_64.egg/sqlalchemy/pool.py", line 651, in __connect
    connection = pool._invoke_creator(self)
  File "/usr/local/lib/python3.6/site-packages/SQLAlchemy-1.1.14-py3.6-linux-x86_64.egg/sqlalchemy/engine/strategies.py", line 105, in connect
    return dialect.connect(*cargs, **cparams)
  File "/usr/local/lib/python3.6/site-packages/SQLAlchemy-1.1.14-py3.6-linux-x86_64.egg/sqlalchemy/engine/default.py", line 393, in connect
    return self.dbapi.connect(*cargs, **cparams)
  File "/usr/local/lib/python3.6/site-packages/py_postgresql-1.2.1-py3.6-linux-x86_64.egg/postgresql/driver/dbapi20.py", line 409, in connect
    return driver.connect(**params)
  File "/usr/local/lib/python3.6/site-packages/py_postgresql-1.2.1-py3.6-linux-x86_64.egg/postgresql/driver/pq3.py", line 3037, in connect
    c.connect()
  File "/usr/local/lib/python3.6/site-packages/py_postgresql-1.2.1-py3.6-linux-x86_64.egg/postgresql/driver/dbapi20.py", line 354, in connect
    super().connect(*args, **kw)
  File "/usr/local/lib/python3.6/site-packages/py_postgresql-1.2.1-py3.6-linux-x86_64.egg/postgresql/driver/pq3.py", line 2430, in connect
    self.close()
  File "/usr/local/lib/python3.6/site-packages/py_postgresql-1.2.1-py3.6-linux-x86_64.egg/postgresql/driver/dbapi20.py", line 363, in close
    creator = self
sqlalchemy.exc.DBAPIError: (postgresql.exceptions.Error) connection already closed
  LOCATION: CLIENT
CONNECTION: [failed]
  failures[0]:
    SSL socket('10.107.62.14', 5432)
    Traceback (most recent call last):
      File "/usr/local/lib/python3.6/site-packages/SQLAlchemy-1.1.14-py3.6-linux-x86_64.egg/sqlalchemy/pool.py", line 1122, in _do_get
        return self._pool.get(wait, self._timeout)
      File "/usr/local/lib/python3.6/site-packages/SQLAlchemy-1.1.14-py3.6-linux-x86_64.egg/sqlalchemy/util/queue.py", line 145, in get
        raise Empty
    sqlalchemy.util.queue.Empty

    During handling of the above exception, another exception occurred:

    Traceback (most recent call last):
      File "/usr/local/lib/python3.6/site-packages/py_postgresql-1.2.1-py3.6-linux-x86_64.egg/postgresql/protocol/client3.py", line 184, in connect
        self.socket = self.socket_factory.secure(self.socket)
      File "/usr/local/lib/python3.6/site-packages/py_postgresql-1.2.1-py3.6-linux-x86_64.egg/postgresql/python/socket.py", line 56, in secure
        return ssl.wrap_socket(socket, **self.socket_secure)
      File "/usr/local/lib/python3.6/site-packages/gevent-1.1.1-py3.6-linux-x86_64.egg/gevent/_ssl3.py", line 576, in wrap_socket
        ciphers=ciphers)
      File "/usr/local/lib/python3.6/site-packages/gevent-1.1.1-py3.6-linux-x86_64.egg/gevent/_ssl3.py", line 126, in __init__
        self._context.verify_mode = cert_reqs
      File "/usr/local/lib/python3.6/ssl.py", line 485, in verify_mode
        super(SSLContext, SSLContext).verify_mode.__set__(self, value)
      File "/usr/local/lib/python3.6/ssl.py", line 485, in verify_mode
        super(SSLContext, SSLContext).verify_mode.__set__(self, value)
      File "/usr/local/lib/python3.6/ssl.py", line 485, in verify_mode
        super(SSLContext, SSLContext).verify_mode.__set__(self, value)
      [Previous line repeated 300 more times]
    RecursionError: maximum recursion depth exceeded

    The above exception was the direct cause of the following exception:

    postgresql.exceptions.InsecurityError: SSL negotiation caused exception
      CODE: --SEC
      LOCATION: CLIENT
CONNECTOR: [Host] pq://postgres:***@postgres:5432
  category: None
DRIVER: postgresql.driver.pq3.Driver
[2017-11-09 17:57:15 +0000] [16] [INFO] Worker exiting (pid: 16)
[2017-11-09 17:57:15 +0000] [1] [INFO] Shutting down: Master
[2017-11-09 17:57:15 +0000] [1] [INFO] Reason: Worker failed to boot.

```

### PR DESCRIPTION
**Pull request by alexandre-vasc**
All CRUD endpoints should refuse inputs bigger than the values defined on database.inputConf.
Added some automated tests related to input checks (edge cases).
Also, Auth now verify who is deleting a user. Auth will no longer permit a user to delete itself. 

Updated Python to 3.6 and gevent to 1.2.2. This should fix #14 

This PR closes dojot/dojot#115 and dojot/dojot#116

### CODE REVIEW

File: auth/database/historicModels.py [hunk 1]
```diff
@@ -18,10 +19,10 @@ class UserInactive(db.Model):
...
 
     id = Column(Integer, primary_key=True, autoincrement=False)
-    name = Column(String, nullable=False)
-    username = Column(String(50), nullable=False)
-    service = Column(String, nullable=False)
-    email = Column(String, nullable=False, unique=True)
+    name = Column(String(UserLimits.name), nullable=False)
+    username = Column(String(UserLimits.username), nullable=False)
+    service = Column(String(UserLimits.service), nullable=False)
+    email = Column(String(UserLimits.email), nullable=False, unique=False)
```
Comment (mmagr) [inline auth/database/historicModels.py]: why drop the constraint?

File: auth/controller/AuthenticationController.py [hunk 2]
```diff
@@ -81,3 +81,9 @@ def getJwtPayload(rawJWT):
         except (jwt.exceptions.DecodeError, sqlalchemy.orm.exc.NoResultFound):
             raise HTTPRequestError(401, "Invalid JWT signaure")
     return jwtPayload
+
+
+def userIdFromJWT(token):
+    if not token:
+        raise HTTPRequestError(401, "not authorized")
+    return getJwtPayload(token[7:])['userid']
```
Comment (mmagr) [inline auth/controller/AuthenticationController.py]: This will only work if the caller supplies a nicely written `bearer ` (space included) won't it?

If we're to ignore the `bearer`part, this should be safer:
```python
return getJwtPayload(token.split(' ')[1])['userid']
```

If not, perhaps use a regexp to filter out the token from the header?
```python
import re
raw_token = re.search('^bearer[ ]+(.*)$', token, flags=re.IGNORECASE).group(1)
return getJwtPayload(raw_token)['userid']
```

File: auth/database/historicModels.py (hunk 1 above)
Comment (alexandre-vasc) [reply auth/database/historicModels.py]: If the automated test script is ran twice, some users are created and deleted twice.
The second user can be created, but the unique constraint on the e-mail field on the inactive table will deny the deletion of the user.
 
if we want unique e-mail and username across active and inactive users, uniqueness checks 
must be done on both tables on user creation. Also, a 'restore inactive user' feature is necessary to reuse a user e-mail.

Anyway, I think those uniqueness checks should be added in another PR.

File: auth/controller/AuthenticationController.py (hunk 2 above)
Comment (alexandre-vasc) [reply auth/controller/AuthenticationController.py]: Done.
I have add a length check to avoid IndexError exception on bearer-less JWT.

File: auth/database/historicModels.py (hunk 1 above)
Comment (mmagr) [reply auth/database/historicModels.py]: mmm :thinking:  

I have a feeling we should discuss the particulars of why we're keeping this users history.
If that is just for logging purposes, perhaps keeping that information on a relational database seems like throwing too much resources at a lesser problem. If not, then those checks definitely need to be implemented. 

That said, I agree with you, we can leave that for later. I've created a new issue (#17) to track that up.
